test/e2e/v4/generate_test.go

```diff
@@ -63,52 +63,7 @@ func GenerateV4(kbc *utils.TestContext) {
 	ExpectWithOffset(1, pluginutil.UncommentCode(
 		filepath.Join(kbc.Dir, "config", "default", "kustomization.yaml"),
 		"#- path: webhookcainjection_patch.yaml", "#")).To(Succeed())
-	ExpectWithOffset(1, pluginutil.UncommentCode(
-		filepath.Join(kbc.Dir, "config", "default", "kustomization.yaml"),
-		"#- metrics_service.yaml", "#")).To(Succeed())
-	ExpectWithOffset(1, pluginutil.UncommentCode(
-		filepath.Join(kbc.Dir, "config", "default", "kustomization.yaml"),
-		metricsTarget, "#")).To(Succeed())
-
-	ExpectWithOffset(1, pluginutil.UncommentCode(filepath.Join(kbc.Dir, "config", "default", "kustomization.yaml"),
-		certManagerTarget, "#")).To(Succeed())
-
-	if kbc.IsRestricted {
-		By("uncomment kustomize files to ensure that pods are restricted")
-		uncommentPodStandards(kbc)
-	}
-}
-
-// GenerateV4WithoutMetrics implements a go/v4 plugin project defined by a TestContext.
-func GenerateV4WithoutMetrics(kbc *utils.TestContext) {
```

Comment:
I think would be great we keep a test to ensure that the metrics are not exposed and we have an option to opt-out from it that works.